NEWS.md

```diff
@@ -1,6 +1,22 @@
 
 # vctrs (development version)
 
+* `vec_cbind()` now calls `vec_restore()` on inputs emptied of their
+  columns. This happens before taking the common type. This has
+  consequences for data frame classes with special columns that
+  devolve into simpler classes when the columns are subsetted
+  out. These classes are now always simplified by `vec_cbind()`.
+
+  For instance, column-binding a grouped data frame with a data frame
```

Comment:
Thinking purely in terms of dplyr, this doesn't feel right to me, because it doesn't seem like adding columns to the RHS should change the grouping defined by the LHS. 

Crazy thought: could we say the the class of the output is defined purely by the LHS? (This might end up being more consistent with dplyr joins)